src/VisualStudio/Core/SolutionExplorerShim/DiagnosticItem/BaseDiagnosticItem.cs

```diff
@@ -8,86 +8,47 @@
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CodeFixes.Configuration;
 using Microsoft.CodeAnalysis.Diagnostics;
-using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
 using Microsoft.VisualStudio.Imaging;
 using Microsoft.VisualStudio.Imaging.Interop;
 using Microsoft.VisualStudio.LanguageServices.Implementation.Utilities;
-using Roslyn.Utilities;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.SolutionExplorer
 {
     internal abstract partial class BaseDiagnosticItem : BaseItem
     {
-        protected readonly DiagnosticDescriptor _descriptor;
-        protected ReportDiagnostic _effectiveSeverity;
+        public string Language { get; }
+        public DiagnosticDescriptor Descriptor { get; }
+        public ReportDiagnostic EffectiveSeverity { get; private set; }
 
         public override event PropertyChangedEventHandler PropertyChanged;
 
-
-        public BaseDiagnosticItem(DiagnosticDescriptor descriptor, ReportDiagnostic effectiveSeverity)
-            : base(string.Format("{0}: {1}", descriptor.Id, descriptor.Title))
+        public BaseDiagnosticItem(DiagnosticDescriptor descriptor, ReportDiagnostic effectiveSeverity, string language)
+            : base(descriptor.Id + ": " + descriptor.Title)
```

Comment:
```suggestion
            : base($"{descriptor.Id}: {descriptor.Title}")
```

Reply:
Interpolated string is less efficient and doesn't make it more readable.